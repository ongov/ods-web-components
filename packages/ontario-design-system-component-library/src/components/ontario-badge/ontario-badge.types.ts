export const BadgeColours = ['teal', 'light-teal', 'black', 'grey', 'red', 'yellow', 'green'] as const;

export enum BadgeColourToClass {
	teal = 'ontario-badge--default-heavy',
	'light-teal' = 'ontario-badge--default-light',
	black = 'ontario-badge--neutral-heavy',
	grey = 'ontario-badge--neutral-light',
	red = 'ontario-badge--alert-heavy',
	yellow = 'ontario-badge--warning-heavy',
	green = 'ontario-badge--success-heavy',
}

export type BadgeColour = (typeof BadgeColours)[number];
