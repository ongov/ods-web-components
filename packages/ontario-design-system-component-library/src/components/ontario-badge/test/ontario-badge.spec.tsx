import { newSpecPage } from '@stencil/core/testing';
import { OntarioBadge } from '../ontario-badge';

describe('ontario-badge', () => {
	describe('snapshot', () => {
		it('should render the expected base html', async () => {
			const page = await newSpecPage({
				components: [OntarioBadge],
				html: `<ontario-badge>In progress</ontario-badge>`,
			});

			expect(page.root).toMatchSnapshot();
		});
	});

	describe('snapshot', () => {
		it('should override the host textContent with label prop output on the span element', async () => {
			const page = await newSpecPage({
				components: [OntarioBadge],
				html: `<ontario-badge label="completed">In progress</ontario-badge>`,
			});

			expect(page.root).toMatchSnapshot();
		});
	});

	describe('snapshot', () => {
		it('should render with the ontario-badge--default-light class on the span element', async () => {
			const page = await newSpecPage({
				components: [OntarioBadge],
				html: `<ontario-badge colour="light-teal">In progress</ontario-badge>`,
			});

			expect(page.root).toMatchSnapshot();
		});
	});

	describe('snapshot', () => {
		it('should render the aria-label attribute on the span element', async () => {
			const page = await newSpecPage({
				components: [OntarioBadge],
				html: `<ontario-badge aria-label-text="This course is currently in progress.">In progress</ontario-badge>`,
			});

			expect(page.root).toMatchSnapshot();
		});
	});
});
