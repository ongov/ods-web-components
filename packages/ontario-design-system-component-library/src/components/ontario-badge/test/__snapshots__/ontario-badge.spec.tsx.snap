// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ontario-badge should override the host textContent with label prop output on the span element 1`] = `
<ontario-badge label="completed">
  <mock:shadow-root>
    <span class="ontario-badge ontario-badge--default-heavy">
      completed
    </span>
  </mock:shadow-root>
  In progress
</ontario-badge>
`;

exports[`ontario-badge should render the aria-label attribute on the span element 1`] = `
<ontario-badge aria-label-text="This course is currently in progress.">
  <mock:shadow-root>
    <span aria-label="This course is currently in progress." class="ontario-badge ontario-badge--default-heavy">
      In progress
    </span>
  </mock:shadow-root>
  In progress
</ontario-badge>
`;

exports[`ontario-badge should render the expected base html 1`] = `
<ontario-badge>
  <mock:shadow-root>
    <span class="ontario-badge ontario-badge--default-heavy">
      In progress
    </span>
  </mock:shadow-root>
  In progress
</ontario-badge>
`;

exports[`ontario-badge should render with the ontario-badge--default-light class on the span element 1`] = `
<ontario-badge colour="light-teal">
  <mock:shadow-root>
    <span class="ontario-badge ontario-badge--default-light">
      In progress
    </span>
  </mock:shadow-root>
  In progress
</ontario-badge>
`;
